src/lib/modules/tests/test.js

```diff
@@ -250,6 +232,30 @@ class Test {
     });
   }
 
+  resetEmbarkJS(cb) {
```

Comment:
I think the logic in this function should go in another module, probably the same that handles the current vm.
Especially since I saw that we never reuse `this.vm` anywhere else.

Reply:
Any suggestions on where this could go?

We could require `EmbarkJS`, `web3`, `Web3`, and `IpfsApi` in the VM constructor, but that wouldn't really make sense as VM would be dependent on the objects. When the VM is used in the console, it does not need these objects, so would make little sense to add to the constructor. As these objects are only used in the tests (both `Test` and `SolcTest`, it seems to make sense to have this method in the base `Test` class.  Creating a separate module just for resetting EmbarkJS also seems a bit overkill. What do you think?

As a side note, objects are passed by value in to the VM sandbox and are manipulated inside the VM (using `VM.doEval()`)

Reply:
Well, `web3` and `EmbarkJS` are needed in the console. Currently, I think it's the console that registers those, but in the future, I think it would be better if something else did it. 
For example, when doing `embark test` we currently need to start the console service as it's the one that makes `EmbarkJS` available globally, which is not intuitive.

Anyway, for now, I think it's fine to keep that reset here, but if we end up refactoring how stuff is added to the VM and to the global space, it might make sense to move it.

Reply:
Apologies for the wording, you are right, `web3` and `EmbarkJS` are needed in the console, but not needed on the instantiation of the VM in the way it stands. Instead, they are registered in the VM sandbox using the `runcode:register` event handler from within the `Console` module.

I agree, this could definitely be refactored as it is not intuitive. We could merge `codeRunner` and `VM` classes, and have the `Console` instantiate a `VM` (passing in `EmbarkJS/web3/Web3/IpfsApi` as part of the sandbox to the constructor), eliminating the need for the `Console` to register `EmbarkJS/web3/Web3/IpfsApi` using `runcode:register`. I still don't know where the `resetEmbarkJS()` would go, if not in the `Test` class, but I guess we can further discuss when we cross that bridge :)